features/topologies/service/impl/src/main/java/org/opennms/netmgt/topologies/service/impl/OnmsTopologyDaoInMemoryImpl.java

```diff
@@ -107,26 +104,23 @@ public void unregister(OnmsTopologyUpdater updater) throws OnmsTopologyException
     }
 
     @Override
-    public void update(OnmsTopologyUpdater updater,
-            OnmsTopologyMessage message) throws OnmsTopologyException {
+    public void update(OnmsTopologyUpdater updater, OnmsTopologyMessage message) {
         final OnmsTopologyProtocol protocol = updater.getProtocol();
         if (!m_updatersMap.containsKey(protocol)) {
-            throw new OnmsTopologyException(String.format("cannot update message with id: %s. Protocol not supported",
-                                                          message.getMessagebody().getId()),
-                                            protocol,
-                                            message.getMessagestatus());
+            throw new IllegalArgumentException(String.format("cannot update message with id: %s. Protocol %s not " +
+                            "supported for message status %s", message.getMessagebody().getId(), protocol,
+                    message.getMessagestatus()));
         }
         if (m_updatersMap.get(protocol) != updater) {
-            throw new OnmsTopologyException(String.format("cannot update message with id: %s. Updater not registered",
-                                                          message.getMessagebody().getId()),
-                                            protocol,
-                                            message.getMessagestatus());
+            throw new IllegalArgumentException(String.format("cannot update message with id: %s, protocol: %s and " +
+                            "message status: %s. Updater not registered", message.getMessagebody().getId(), protocol,
+                    message.getMessagestatus()));
         }
         synchronized (m_consumers) {
             m_consumers
-                .stream()
-                .filter(consumer ->  consumer.getProtocols().contains(protocol))
-                .forEach(consumer -> consumer.consume(message));            
+                    .stream()
+                    .filter(consumer -> consumer.getProtocols().contains(protocol))
```

Comment:
Is there a case, where `consumer.getProtocols()` can be null?

Reply:
There isn't currently, but the method is not documented to imply null is not a valid return value, therefore I'll add null handling here.